### PR TITLE
Optimize TimeoutEntity indexes in custom schemas

### PR DESCRIPTION
**Pull request by MarcinHoppe**

### CODE REVIEW

**Comment by MarcinHoppe** [comment]
@SzymonPobiega Can you have a look at this PR?

File: src/NServiceBus.NHibernate/TimeoutPersisters/TimeoutEntity.cs [hunk 1]
```diff
@@ -29,18 +29,26 @@ class TimeoutEntity
         public virtual byte[] State { get; set; }
 
         /// <summary>
+        /// Timeout endpoint name.
```
Comment (SzymonPobiega) [inline src/NServiceBus.NHibernate/TimeoutPersisters/TimeoutEntity.cs]: Why do we need *both* the replacement in `OptimizedSchemaUpdated` and these comments? Would one of them not suffice?

Comment (MarcinHoppe) [inline src/NServiceBus.NHibernate/TimeoutPersisters/TimeoutEntity.cs]: This alone will not give us a clustered index on (Endpoint, Time) on SQL Server. It is still needed for proper order column on other databases (read: Oracle).

Comment (SzymonPobiega) [inline src/NServiceBus.NHibernate/TimeoutPersisters/TimeoutEntity.cs]: 👍

**Comment by SzymonPobiega** [comment]
LGTM

**Pull request merged**